contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,792 @@
+    }
+
+    /**
+     * @dev Deletes a series and cleans up related data
+     */
+    function _deleteSeries(uint256 seriesID) 
+        internal 
+        seriesExists(seriesID) 
+        onlyOwnerOrArtist(seriesID) 
+    {
+        // Clean up all pending co-artist requests for this series in one loop
+        uint256[] memory pendingCoArtists = seriesDetails[seriesID].pendingCoArtists;
+        for (uint256 i = 0; i < pendingCoArtists.length; i++) {
+            uint256 artistID = pendingCoArtists[i];
+            seriesPendingCoArtist[seriesID][artistID] = false;
+            _removeArtistPendingCoArtistSeries(artistID, seriesID);
+        }
+
+        Series memory series = seriesDetails[seriesID];
+        
+        // Remove series from all artists
+        for (uint256 i = 0; i < series.artistIDs.length; i++) {
+            _removeSeriesFromArtist(series.artistIDs[i], seriesID);
+        }
+
+        delete seriesDetails[seriesID];
+
+        emit SeriesDeleted(seriesID);
+    }
+
+    /**
+     * @dev Removes a series from an artist's list
+     */
+    function _removeSeriesFromArtist(uint256 artistID, uint256 seriesID) internal {
+        uint256[] storage seriesIDs = artists[artistID].seriesIDs;
+        
+        for (uint256 i = 0; i < seriesIDs.length; i++) {
+            if (seriesIDs[i] == seriesID) {
+                seriesIDs[i] = seriesIDs[seriesIDs.length - 1];
+                seriesIDs.pop();
+                break;
+            }
+        }
+        
+        seriesArtistExist[seriesID][artistID] = false;
+    }
+
+    /**
+     * @dev Validates that artists can only add series that only have themselves as artist
+     */
+    function _validateOnlySelfAsArtist(
+        address[] memory artistAddrs
+    ) internal view {
+        if (artistAddrs.length != 1 || artistAddrs[0] != msg.sender) {
+            revert NotAuthorizedError(msg.sender);
+        }
+    }
+
+    /**
+     * @dev Validates that metadataURI and token URI are not empty
+     */
+    function _validateMetadataAndTokenURI(
+        string memory metadataURI,
+        string memory tokenIDsMapURI
+    ) internal pure {
+        if (bytes(metadataURI).length == 0) {
+            revert EmptyMetadataURIError();
+        }
+        if (bytes(tokenIDsMapURI).length == 0) {
+            revert EmptyTokenMapURIError();
+        }
+    }
+
+    /**
+     * @dev Validates batch parameters for series creation
+     */
+    function _validateBatchParameters(
+        uint256 seriesCount,
+        string[] memory metadataURIs,
+        string[] memory tokenIDsMapURIs
+    ) internal pure {
+        if (seriesCount != metadataURIs.length || seriesCount != tokenIDsMapURIs.length) {
+            revert ArrayLengthMismatchError(
+                seriesCount,
+                metadataURIs.length,
+                tokenIDsMapURIs.length
+            );
+        }
+        if (seriesCount == 0) {
```

Comment:
Should check these count first since it's cheaper. 
And using in-memory variables to avoid access array `length` multiple times, it could be better in readability and gas (not so sure)